lib/ui/window/platform_configuration.h

```diff
@@ -217,6 +217,26 @@ class PlatformConfigurationClient {
   ///
   virtual void RequestDartDeferredLibrary(intptr_t loading_unit_id) = 0;
 
+  //--------------------------------------------------------------------------
+  /// @brief      Synchronously invokes platform-specific APIs to apply the
+  ///             system text scaling on the given unscaled font size.
+  ///
+  /// @param[in]  unscaled_font_size  The unscaled font size specified by the
+  ///                                 app developer. The value is guaranteed
+  ///                                 to be finite and non-negative.
+  /// @param[in]  configuration_id    The unique id of the configuration to use
+  ///                                 for computing the scaled font size.
+  ///
+  /// @return     The scaled font size in logical pixels. Returns a negative
+  ///             number when an error is encountered:
+  ///              - Returns -1 if this method isn't implemented on the current
+  ///                platform.
+  ///              - Returns -2 if the given configuration_id does not match
+  ///                the platform's text scale factor value.
+  ///
+  virtual double GetScaledFontSize(double unscaled_font_size,
+                                   int configuration_id) const = 0;
```

Comment:
configuration_id i assume this is configuration_generation? we should unify the name